src/backend/distributed/metadata/node_metadata.c

```diff
@@ -371,8 +386,12 @@ SetUpDistributedTableDependencies(WorkerNode *newWorkerNode)
 		EnsureNoModificationsHaveBeenDone();
 		ReplicateAllDependenciesToNode(newWorkerNode->workerName,
 									   newWorkerNode->workerPort);
-		ReplicateAllReferenceTablesToNode(newWorkerNode->workerName,
-										  newWorkerNode->workerPort);
+
+		if (ReplicateReferenceTablesOnActivate)
+		{
+			ReplicateAllReferenceTablesToNode(newWorkerNode->workerName,
+											  newWorkerNode->workerPort);
```

Comment:
In a future PR, might be worth exploring whether we can do this by calling master_copy_shard_placement directly (in this backend). That way we can keep one code path. For now that's not terribly important.